autocorrector.py

```diff
@@ -0,0 +1,224 @@
+"""Performs XMAlab-style autocorrection on the trials and videos"""
+
+import math
+import os
+
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from ruamel.yaml import YAML
+
+from xma_data_processor import XMADataProcessor
+
+
+class Autocorrector:
+    """Performs XMAlab-style autocorrection on the trials and videos"""
+
+    def __init__(self, config):
+        self._trials_path = os.path.join(config["working_dir"], "trials")
+        self._data_processor = XMADataProcessor(config)
+        self._config = config
+        self._dlc_config = config['path_config_file']
+
+    def autocorrect_trial(self):
+        '''Do XMAlab-style autocorrect on the tracked beads'''
+        trials = self._data_processor.list_trials()
+
+        # Establish project vars
+        yaml = YAML()
+        with open(self._dlc_config) as dlc_config:
+            dlc = yaml.load(dlc_config)
+
+        iteration = dlc['iteration']
+
+        for trial in trials:
+            # Find the appropriate pointsfile
+            trial_path = os.path.join(self._trials_path, trial)
+            iteration_path = os.path.join(trial_path, f'it{iteration}')
+
+            try:
+                csv = pd.read_csv(os.path.join(iteration_path, f'{trial}-Predicted2DPoints.csv'))
+            except FileNotFoundError:
```

Comment:
This is not great because it creates 2 FileNotFoundError things on the stack. Change to
```python
except FileNotFoundError as e:
    print('some-error-message')
    raise e
```

Reply:
done